### ISSUE TITLE
istioctl proxy-status returns 404 when executed for specific pod

### ISSUE THREAD

**Issue by himanshujaindev** [istio/istio#49693]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

```
$ istio-1.20.3/bin/istioctl proxy-status istio-ingressgateway-7857fb9bd5-89j2m.istio-system
Error: unexpected status code: 404
```

Please help me understand why this error occurs and how can I correct it.

### Version

```Text
$ istio-1.20.3/bin/istioctl version
client version: 1.20.3
control plane version: 1.20.2
data plane version: 1.20.2 (10 proxies)
```

### Additional Information

CLI version:
```
version.BuildInfo{Version:"1.20.3", GitRevision:"692e556046b48ebc471205211c68a2c69e74a321", GolangVersion:"go1.21.6", BuildStatus:"Clean", GitTag:"1.20.3"}
```

The following Istio control plane revisions/versions were found in the cluster:
Revision 1-20-2:
```
&version.MeshInfo{
    {
        Component: "istiod",
        Revision:  "1-20-2",
        Info:      version.BuildInfo{Version:"1.20.2", GitRevision:"5f5d657c72d30a97cae97938de3a6831583e9f15", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.20.2"},
    },
    {
        Component: "istiod",
        Revision:  "1-20-2",
        Info:      version.BuildInfo{Version:"1.20.2", GitRevision:"5f5d657c72d30a97cae97938de3a6831583e9f15", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.20.2"},
    },
    {
        Component: "istiod",
        Revision:  "1-20-2",
        Info:      version.BuildInfo{Version:"1.20.2", GitRevision:"5f5d657c72d30a97cae97938de3a6831583e9f15", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.20.2"},
    },
}
```

**Comment by howardjohn** [comment]
Adding `--log_output_level=all:debug --vklog=6` maybe gives more info

**Comment by nhimanshujain** [comment]
```
$ istio-1.20.3/bin/istioctl proxy-status istio-ingressgateway-7857fb9bd5-89j2m.istio-system --log_output_level=all:debug --vklog=6
2024-03-05T15:28:47.812155Z	debug	klog	Config loaded from file:  <redacted>
2024-03-05T15:28:47.822109Z	debug	klog	Config loaded from file:  <redacted>
2024-03-05T15:28:48.866314Z	info	klog	GET https://<redacted>/api/v1/namespaces/istio-system/pods/istio-ingressgateway-7857fb9bd5-89j2m 200 OK in 1042 milliseconds
2024-03-05T15:28:49.290407Z	info	klog	GET https://<redacted>/api/v1/namespaces/istio-system/pods/istio-ingressgateway-7857fb9bd5-89j2m 200 OK in 252 milliseconds
2024-03-05T15:28:50.165034Z	info	klog	POST https://<redacted>/api/v1/namespaces/istio-system/pods/istio-ingressgateway-7857fb9bd5-89j2m/portforward 101 Switching Protocols in 872 milliseconds
2024-03-05T15:28:50.165918Z	debug	Port forward established localhost:56587 -> istio-ingressgateway-7857fb9bd5-89j2m.istio-ingressgateway-7857fb9bd5-89j2m:15000
2024-03-05T15:28:53.573904Z	info	port forward completed without error
2024-03-05T15:28:53.824756Z	info	klog	GET https://<redacted>/api/v1/namespaces/istio-system/pods?fieldSelector=status.phase%3DRunning&labelSelector=app%3Distiod 200 OK in 249 milliseconds
2024-03-05T15:28:54.324128Z	info	klog	GET https://<redacted>/api/v1/namespaces/istio-system/pods/istiod-1-20-2-68cc8cd877-8sbxr 200 OK in 251 milliseconds
2024-03-05T15:28:55.091831Z	info	klog	POST https://<redacted>/api/v1/namespaces/istio-system/pods/istiod-1-20-2-68cc8cd877-8sbxr/portforward 101 Switching Protocols in 766 milliseconds
2024-03-05T15:28:55.092128Z	debug	Port forward established localhost:56593 -> istiod-1-20-2-68cc8cd877-8sbxr.istiod-1-20-2-68cc8cd877-8sbxr:15014
Error: unexpected status code: 404
```

@howardjohn - Providing the requested logs for more details

**Comment by howardjohn** [comment]
This is doing `curl localhost:15014/debug/syncz` basically and getting 404. Do you have `--monitoringAddr` overriden or otherwise turning off these APIs?

**Comment by nhimanshujain** [comment]
No. I have not overridden the variable. Just in case, how can I verify the same?

**Comment by howardjohn** [comment]
I would try manually doing that curl in an istiod pod to verify I am correct. Then paste the full istiod logs maybe (you can restart it to make the logs pretty short, we just need the startup logs)?

**Comment by nhimanshujain** [comment]
```
$ k exec istiod-1-20-2-8459478499-czpcx -n istio-system -c discovery -it -- bin/bash
istio-proxy@istiod-1-20-2-8459478499-czpcx:/$ curl 'localhost:15014/debug/syncz'
[{"cluster_id":<redacted>,"proxy":<redacted>,"proxy_type":"sidecar","istio_version":"1.20.2","cluster_sent":"95482258-de24-4f34-83f1-9bc9c23ff205","cluster_acked":"95482258-de24-4f34-83f1-9bc9c23ff205","listener_sent":"fdfb88f2-7b76-4be5-abad-6332f79ef5ae","listener_acked":"fdfb88f2-7b76-4be5-abad-6332f79ef5ae","route_sent":"2d013ece-e06b-4adb-9ca0-0cc771c17853","route_acked":"2d013ece-e06b-4adb-9ca0-0cc771c17853","endpoint_sent":"2ea57c63-342b-41f0-8504-89faae80cd87","endpoint_acked":"2ea57c63-342b-41f0-8504-89faae80cd87"}]
```

I restarted istiod pods and when I look for logs during the execution of above command, there are none present.
FYI: Used this command to get the logs
```
$ k logs -l app=istiod -c discovery -n istio-system --tail 1 -f
```

**Comment by howardjohn** [comment]
I think we might be seeing this here:

https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/49890/integ-cni_istio/1768226937248944128/build-log.txt

`2024-03-14T10:53:16.351777Z	warn	tf	failed dumping configz (cluster cluster-0): invalid character 'N' after top-level value`

Presumably 'N' is "**N**ot found"?

**Comment by howardjohn** [comment]
that was https://github.com/istio/istio/pull/49921, I think its unrelated.

**Comment by nhimanshujain** [comment]
@howardjohn - Is this a bug which needs to be addressed ?

**Comment by howardjohn** [comment]
perhaps, it's unclear what the root cause is so unclear if it's a bug.

**Comment by tchellomello** [comment]
Hitting the same issue: 

```yaml
❯ istioctl proxy-status | grep details-v1
details-v1-86867d5bdc-5x8dc.bookinfo                                   Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT SENT     istiod-69d98ff48d-l4bcc     1.21.0

❯ istioctl proxy-status pod/details-v1-86867d5bdc-5x8dc.bookinfo --log_output_level=all:debug --vklog=6
2024-04-08T03:49:53.483649Z	debug	klog	Config loaded from file:  /home/mmello/.kube/config
2024-04-08T03:49:53.496065Z	debug	klog	Config loaded from file:  /home/mmello/.kube/config
2024-04-08T03:49:53.641372Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/bookinfo/pods/details-v1-86867d5bdc-5x8dc 200 OK in 135 milliseconds
2024-04-08T03:49:53.646640Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/bookinfo/pods/details-v1-86867d5bdc-5x8dc 200 OK in 4 milliseconds
2024-04-08T03:49:53.653455Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/bookinfo/pods/details-v1-86867d5bdc-5x8dc 200 OK in 6 milliseconds
2024-04-08T03:49:53.674265Z	info	klog	POST https://192.168.111.201:443/api/v1/namespaces/bookinfo/pods/details-v1-86867d5bdc-5x8dc/portforward 101 Switching Protocols in 20 milliseconds
2024-04-08T03:49:53.674464Z	debug	Port forward established localhost:43891 -> details-v1-86867d5bdc-5x8dc.details-v1-86867d5bdc-5x8dc:15000
2024-04-08T03:49:54.290943Z	info	port forward completed without error
2024-04-08T03:49:54.300738Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/istio-system/pods?fieldSelector=status.phase%3DRunning&labelSelector=app%3Distiod 200 OK in 9 milliseconds
2024-04-08T03:49:54.306826Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/istio-system/pods/istiod-69d98ff48d-b85vd 200 OK in 3 milliseconds
2024-04-08T03:49:54.384562Z	info	klog	POST https://192.168.111.201:443/api/v1/namespaces/istio-system/pods/istiod-69d98ff48d-b85vd/portforward 101 Switching Protocols in 77 milliseconds
2024-04-08T03:49:54.384727Z	debug	Port forward established localhost:40559 -> istiod-69d98ff48d-b85vd.istiod-69d98ff48d-b85vd:15014
Error: unexpected status code: 404
```

**Comment by tchellomello** [comment]
@hanxiaop thanks for your input. 

I've updated for `1.21.1` today and it is a bit different: 


```bash
❯ istioctl  x ps | grep reviews
reviews-v1-69dbff5b64-kqsh6.bookinfo                                   Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT_SENT     istiod-66fcf48747-dgtmr     1.21.1
reviews-v2-6cbbd5c5bb-lghft.bookinfo                                   Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT_SENT     istiod-66fcf48747-dgtmr     1.21.1
reviews-v3-64df9b4b48-4stcn.bookinfo                                   Kubernetes     SYNCED     SYNCED     SYNCED     SYNCED     NOT_SENT     istiod-66fcf48747-lk5q9     1.21.1


❯ istioctl  x ps reviews-v2-6cbbd5c5bb-lghft.bookinfo 
Clusters Match
Listeners Match
Routes Match (RDS last loaded at Tue, 09 Apr 2024 01:11:20 EDT)

❯ istioctl  x ps reviews-v2-6cbbd5c5bb-lghft.bookinfo --log_output_level=all:debug --vklog=6  
2024-04-09T05:15:44.399735Z	debug	klog	Config loaded from file:  /home/mmello/.kube/config
2024-04-09T05:15:44.419817Z	debug	klog	Config loaded from file:  /home/mmello/.kube/config
2024-04-09T05:15:44.535675Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/bookinfo/pods/reviews-v2-6cbbd5c5bb-lghft 200 OK in 114 milliseconds
2024-04-09T05:15:44.539507Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/bookinfo/pods/reviews-v2-6cbbd5c5bb-lghft 200 OK in 3 milliseconds
2024-04-09T05:15:44.564008Z	info	klog	POST https://192.168.111.201:443/api/v1/namespaces/bookinfo/pods/reviews-v2-6cbbd5c5bb-lghft/portforward 101 Switching Protocols in 23 milliseconds
2024-04-09T05:15:44.564219Z	debug	Port forward established localhost:33933 -> reviews-v2-6cbbd5c5bb-lghft.reviews-v2-6cbbd5c5bb-lghft:15000
2024-04-09T05:15:45.065524Z	info	port forward completed without error
2024-04-09T05:15:45.073967Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/istio-system/pods?fieldSelector=status.phase%3DRunning&labelSelector=app%3Distiod 200 OK in 8 milliseconds
2024-04-09T05:15:45.080963Z	info	klog	POST https://192.168.111.201:443/api/v1/namespaces/istio-system/serviceaccounts/default/token 201 Created in 6 milliseconds
2024-04-09T05:15:45.083996Z	info	klog	GET https://192.168.111.201:443/api/v1/namespaces/istio-system/pods/istiod-66fcf48747-dgtmr 200 OK in 2 milliseconds
2024-04-09T05:15:45.141334Z	info	klog	POST https://192.168.111.201:443/api/v1/namespaces/istio-system/pods/istiod-66fcf48747-dgtmr/portforward 101 Switching Protocols in 57 milliseconds
2024-04-09T05:15:45.141493Z	debug	Port forward established localhost:38635 -> istiod-66fcf48747-dgtmr.istiod-66fcf48747-dgtmr:15012
2024-04-09T05:15:45.162556Z	debug	adsc	Sending Discovery Request to istiod: {
  "node": {
    "id": "sidecar~192.168.111.36~test-1.default~default.svc.cluster.local",
    "metadata": {
        "GENERATOR": "event",
        "ISTIO_VERSION": "65536.65536.65536",
        "NAMESPACE": "istio-system",
        "SERVICE_ACCOUNT": "default"
      }
  },
  "resourceNames": [
    "reviews-v2-6cbbd5c5bb-lghft.bookinfo"
  ],
  "typeUrl": "istio.io/debug/config_dump",
  "responseNonce": "2024-04-09 01:15:45.162293546 -0400 EDT m=+0.818568411"
}
2024-04-09T05:15:45.249598Z	info	adsc	Received	type=istio.io/debug/config_dump count=7 nonce=b0778970-5fb4-4d07-a943-dfb7c194e8de
2024-04-09T05:15:45.249893Z	info	port forward completed without error
2024-04-09T05:15:45.250038Z	info	adsc	Connection closed for node sidecar~192.168.111.36~test-1.default~default.svc.cluster.local with err: rpc error: code = Unavailable desc = error reading from server: EOF
Clusters Match
Listeners Match
Routes Match (RDS last loaded at Tue, 09 Apr 2024 01:11:20 EDT)
```

**Comment by taisph** [comment]
FWIW, I am getting this on istio 1.19.10 and x ps results in a timeout.
